lib/agent.js

```diff
@@ -250,33 +247,43 @@ Agent.prototype.start = function (opts) {
     return this
   }
 
-  this._origStackTraceLimit = Error.stackTraceLimit
-  Error.stackTraceLimit = 20 // require more than default 10 for `agentActivationMethodFromStartStack()`
+  // Log preamble showing agent, environment and config relevant info
+  const preambleData = this._conf.loggingPreambleData
+  const isPreviewVersion = version.indexOf('-') !== -1
   const startStack = {}
-  Error.captureStackTrace(startStack)
-  this._agentActivationMethod = agentActivationMethodFromStartStack(startStack, this.logger)
-
-  if (this._conf.logLevel === 'trace') {
-    // Attempt to load package.json from process.argv.
-    var pkg = null
-    try {
-      var basedir = path.dirname(process.argv[1] || '.')
-      pkg = require(path.join(basedir, 'package.json'))
-    } catch (e) {}
-
-    this.logger.trace({
-      pid: process.pid,
-      ppid: process.ppid,
-      arch: process.arch,
-      platform: process.platform,
-      node: process.version,
-      agent: version,
-      agentActivationMethod: this._agentActivationMethod,
-      startTrace: startStack.stack.split(/\n */).slice(1),
-      main: pkg ? pkg.main : '<could not determine>',
-      dependencies: pkg ? pkg.dependencies : '<could not determine>',
-      conf: this._conf.toJSON()
-    }, 'agent configured correctly')
+
+  if (this._conf.active && this._conf.serviceName) {
+    this._origStackTraceLimit = Error.stackTraceLimit
+    Error.stackTraceLimit = 20 // require more than default 10 for `agentActivationMethodFromStartStack()`
+    Error.captureStackTrace(startStack)
+    this._agentActivationMethod = agentActivationMethodFromStartStack(startStack, this.logger)
+    preambleData.activationMethod = this._agentActivationMethod
+
+    if (this._conf.logLevel === 'trace') {
+      // Attempt to load package.json from process.argv.
+      let pkg = null
+      try {
+        var basedir = path.dirname(process.argv[1] || '.')
+        pkg = require(path.join(basedir, 'package.json'))
+      } catch (e) {}
```

Comment:
Side note: This only looks in the same dir as the script. I think it might be fairly common for scripts to be in a "scripts" or "bin" or other subdir. Looking up at least one dir might be interesting. For example `ELASTIC_APM_LOG_LEVEL=trace node examples/trace-http.js` results in `pkg` not being found.   This isn't something for *this* PR though.